docs/source/work-stealing.rst

```diff
@@ -0,0 +1,86 @@
+Work Stealing
+=============
+
+Some tasks prefer to run on certain workers.  This may be because that worker
+holds data dependencies of the task or because the user has expressed a loose
+desire that the task run in a particular place.  Occasionally this results in a
+few very busy workers and several idle workers.  In this situation the idle
+workers may choose to steal work from busy workers, even if stealing work
+requires the costly movement of data.
+
+This is a performance optimization and not required for correctness.  Work
+stealing provides robustness in many ad-hoc cases, but can also backfire when
+we steal the wrong tasks and reduce performance.
+
+
+Task criteria for stealing
+--------------------------
+
+If a task has been specifically restricted to run on particular workers (such
+as is the case when special hardware is required) then we do not steal.
```

Comment:
It's not prohibited, just not implemented.  Generally there is a high cost for adding any logic here (error prone, very performance sensitive) so I've tended to avoid some of the more fringe cases.  They're definitely valid, they just haven't shown yet themselves to be worth the development and maintenance cost.